From 7a5f23589fde3e6d5394cedda3f43990d891f91e Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Mon, 9 Jul 2018 19:27:44 +0800
Subject: [PATCH 4134/5242] MLK-18816-2 input/touch: do not clear touch
 interrupt when enable irq

commit  1102c98ca117557212d8ef8c6de2d71dbba84aef from
https://source.codeaurora.org/external/imx/linux-imx.git

On imx8mscale-evk baord, if I2C bus is configed pull-up, then once send
the i2c command to clear touch interrupt during the touch initialization,
touch will keep SDA line in low level, block the i2c bus. If config the
I2C bus pull-down, then this issue gone. Due to it is not reasonable to
set the I2c bus to pull-down for other i2c slave device, this patch work
as a workaround, just remove this i2c command, do not clear the touch
interrupt, test that touch can also work well after the initialization.

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../touchscreen/synaptics_dsx/synaptics_dsx_i2c.c  |   10 ----------
 1 file changed, 10 deletions(-)

diff --git a/drivers/input/touchscreen/synaptics_dsx/synaptics_dsx_i2c.c b/drivers/input/touchscreen/synaptics_dsx/synaptics_dsx_i2c.c
index b05e561..415ea94 100644
--- a/drivers/input/touchscreen/synaptics_dsx/synaptics_dsx_i2c.c
+++ b/drivers/input/touchscreen/synaptics_dsx/synaptics_dsx_i2c.c
@@ -1618,18 +1618,8 @@ static int synaptics_rmi4_irq_enable(struct synaptics_rmi4_data *rmi4_data,
 		bool enable)
 {
 	int retval = 0;
-	unsigned char intr_status[MAX_INTR_REGISTERS];
 
 	if (enable) {
-
-		/* Clear interrupts first */
-		retval = synaptics_rmi4_i2c_read(rmi4_data,
-				rmi4_data->f01_data_base_addr + 1,
-				intr_status,
-				rmi4_data->num_of_intr_regs);
-		if (retval < 0)
-			return retval;
-
 		/* set up irq */
 		if (!rmi4_data->irq_enabled) {
 #ifdef CONFIG_OF_TOUCH
-- 
1.7.9.5

